bridges/snowbridge/primitives/core/src/fees.rs

```diff
@@ -0,0 +1,42 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2023 Snowfork <hello@snowfork.com>
+use log;
+use sp_runtime::{DispatchResult, SaturatedConversion, Saturating, TokenError};
+use xcm::opaque::latest::{Location, XcmContext};
+use xcm_executor::traits::TransactAsset;
+const LOG_TARGET: &str = "xcm_fees";
+
+/// Burns the fees embedded in the XCM for teleports.
+pub fn burn_fees<AssetTransactor, Balance>(dest: Location, fee: Balance) -> DispatchResult
```

Comment:
this is not used anywhere in this PR

you should call it in `inbound_queue_v2::submit()` before or within [Self::send_xcm()](https://github.com/claravanstaden/polkadot-sdk/blob/687e45e5661633bd8631084e085c48f4be9e0942/bridges/snowbridge/pallets/inbound-queue-v2/src/lib.rs#L237)

it could also be used in inbound-queue-v1 and you can remove [old fn burn_fees()](https://github.com/claravanstaden/polkadot-sdk/blob/687e45e5661633bd8631084e085c48f4be9e0942/bridges/snowbridge/pallets/inbound-queue/src/lib.rs#L347)

Reply:
Good catch, this was obsolete code. Deleted in https://github.com/paritytech/polkadot-sdk/pull/7402/commits/33f8e6d951c2ec10b1bb5337f61a69dcadbcf008

In V2, we don't need to burn any asset for teleporting. No sovereign accounts, etc.

Reply:
yes, with fees paid exclusively using eth, there are no more teleports involved so no need to locally burn anything